src/tests/Common/CLRTest.Jit.targets

```diff
@@ -365,6 +372,139 @@ set DOTNET_JitPath=%CORE_ROOT%\superpmi-shim-collector.dll
     </PropertyGroup>
   </Target>
 
+  <!--
+  ***********************************************************************************************
+  IlasmRoundTrip for merged tests
+  ***********************************************************************************************
+  -->
+
+  <Target Name="IlasmRoundTripCommand" AfterTargets="AfterBuild" Condition="'$(IsMergedTestRunnerAssembly)' == 'true'">
+    <PropertyGroup>
+      <InputAssemblyName>$(AssemblyName).dll</InputAssemblyName>
+      <DisassemblyName>IL-RT/$(AssemblyName).il</DisassemblyName>
+      <TargetAssemblyName>IL-RT/$(AssemblyName).dll</TargetAssemblyName>
+      <_IlasmRoundTripScriptText><![CDATA[
+import os
+import shutil
+import subprocess
+import sys
+import glob
+import struct
+import platform
+
+def is_managed_assembly(file):
+    file_size = os.path.getsize(file)
+
+    if file_size < 64:
+        return False
+
+    with open(file, "rb") as f:
+        f.seek(0x3c)
+
+        # PE Header starts @ 0x3C (60). Its a 4 byte header.
+        peHeaderPointer = struct.unpack('I', f.read(4))[0]
+        if peHeaderPointer == 0:
+            peHeaderPointer = 0x80
+
+        # Ensure there is at least enough room for the following structures:
+        #     24 byte PE Signature & Header
+        #     28 byte Standard Fields         (24 bytes for PE32+)
+        #     68 byte NT Fields               (88 bytes for PE32+)
+        # >= 128 byte Data Dictionary Table
+        if peHeaderPointer > (file_size - 256):
+            return False
+
+        # Check the PE signature.  Should equal 'PE\0\0'.
+        f.seek(peHeaderPointer)
+        peHeaderSignature = struct.unpack('I', f.read(4))[0]
+        if peHeaderSignature != 0x00004550:
+            return False
+
+        # skip over the PEHeader fields
+        f.seek(f.tell() + 20)
+
+        PE32 = 0x10b
+        PE32Plus = 0x20b
+
+        # Read PE magic number from Standard Fields to determine format.
+        peFormat = struct.unpack('H', f.read(2))[0]
+        if peFormat != PE32 and peFormat != PE32Plus:
+            return False
+
+        # Read the 15th Data Dictionary RVA field which contains the CLI header RVA.
+        # When this is non-zero then the file contains CLI data otherwise not.
+        dataDictionaryStart = (peHeaderPointer + (232 if peFormat == PE32 else 248)) & 0xffff
+        f.seek(dataDictionaryStart)
+
+        cliHeaderRva = struct.unpack('I', f.read(4))[0]
+        if cliHeaderRva == 0:
+            return False
+
+        return True
+
+print("")
+print("ILASM RoundTrips")
+
+if not os.path.exists("IL-RT"):
+  os.mkdir("IL-RT")
+
+if "arm" in platform.machine():
+  # TODO: Fix these roundtrip assemblies from failing on ARM64 when executing.
+  assembliesToIgnore = [
+    "value_numbering_checked_arithmetic_with_constants_ro.dll",
+    "ldc_mul_ovf_i4.dll",
+    "overflow02_mul.dll",
+    "b25815.dll",
+    "b71869.dll"
+  ]
```

Comment:
Does IlasmRoundTripIncompatible still work with this?  This work shouldn't be blocked by failing tests - just disable them - but the mechanism needs to be accessible like IlasmRoundTripIncompatible rather than changing this script.

Reply:
It doesn't and there isn't an easy way to know.

Reply:
After looking at CI, these tests were actually failing in non-ilasm roundtrip runs, so I suspect I will remove this.

Reply:
Wait, isn't this just called from the main .cmd/.sh scripts?  IlasmRoundTripIncompatible would stop the execution up front and never get here.

Reply:
Right, but that only works if the merged test project is `IlasmRoundTripIncompatible`, not the individual test itself unfortunately.

Reply:
Short-term fix for that is that RequiresProcessIsolation should be set.

Reply:
I checked all the tests with `IlasmRoundTripIncomptabiel` and they all have either have OutputType=Exe or RequiresProcessIsolation=true, so we don't have to worry about it for now.

Reply:
Yes, that was done intentionally.  They wouldn't have been skipped before either.  So you can keep using this instead of `assembliesToIgnore` if any are broken and for any future regressions that need disabling.